app/controllers/application_controller.rb

```diff
@@ -27,8 +29,23 @@ def set_user
     @user = current_user
   end
 
+  def maintenance_mode
+    redirect_to '/' and return unless maintenance_mode_on?
+    render layout: nil, file: "layouts/maintenance"
+  end
+
   private
 
+  def check_maintenance_mode
+    if maintenance_mode_on? and request.fullpath != '/maintenance'
+      redirect_to '/maintenance' and return
+    end
+  end
```

Comment:
Replace `if maintenance_mode_on? and` with &&

Reply:
Good spot, thanks! Done